core/state_processor.go

```diff
@@ -86,8 +86,12 @@ func (p *StateProcessor) Process(block *types.Block, statedb *state.StateDB, cfg
 		receipts = append(receipts, receipt)
 		allLogs = append(allLogs, receipt.Logs...)
 	}
+	// Fail if Shanghai not enabled and len(withdrawals) is non-zero.
```

Comment:
Don't we need to add the counter-case too ? I.e: fail if shanghai is enabled and withdrawals is nil ? Or is that handled somewhere else?